### DISCUSSION TITLE
Highlight which block(s) are currently running in block coding

### DISCUSSION THREAD

**Discussion by EmperorKabir** [pybricks/support#1380] [Ideas]
For coding n00bs such as myself, it would be really useful to see what block is currently active when code is running; this will make it a lot easier to debug issues. Eg the outline of the active block(s) becomes bolder and/or text italicised and/or changes colour to an as yet unused colour.

I'm guessing though that this would require some sort of feedback back from the hub to the browser as the code is uploaded to the hub rather than run from the browser?

Even if the coding window to the right highlighted the current line it was on (rather than having to create animations/formatting for the block coding side on the left), this would still be really useful for understanding what line of code it's currently on/stuck on.

Understandable if this is too demanding/not fast enough to feedback though but I thought it was worth suggesting.

**Comment by laurensvalk** [comment]
Thanks for the suggestion! As a first step, I think we'll also want to highlight the block that causes an error, if any.

**Reply by EmperorKabir** [reply]
Hey it's been a few months and I've not being playing with pybricks and might come back to fix my buggy train code lol. Out of interest, has this idea progressed/been picked up at all? @laurensvalk

**Reply by laurensvalk** [reply]
Have you tried adding print blocks throughout your program?

In the most excessive case, adding one print for each normal block, basically achieves the same as highlighting the active block. :smile: But of course you can usually get to what you need with far fewer print blocks.

Automatic highlighting has not been done yet.

**Reply by EmperorKabir** [reply]
Yes I thought that might be a solution but the initial idea would make life much easier! Especially for a code with lots of blocks and bugs!
